list.py

```diff
@@ -0,0 +1,277 @@
+class Node:
+    def __init__(self, elem, next = None):
+        self.elem = elem
+        self.next = next
+
+
+class ListIterator:
+    def __init__(self, head):
+        self.current = head
+
+    
+    def __iter__(self):
+        return self
+
+
+    def __next__(self):
+        if self.current == None:
+            raise StopIteration
+
+        elem = self.current.elem
+        self.current = self.current.next
+        return elem
+
+
+class List:
+    def __init__(self):
+        self.head = None
+        self.tail = None
+        self._length = 0
+
+
+    def __len__(self):
+        return self._length
+
+
+    def length(self) -> int:
+        return self._length
+
+
+    def __eq__(self, other: 'List'):
+        '''
+        pre: 
+            two lists
+        post: 
+            True if lists are equals, False if not
+        '''
+        curr_node = self.head
+
+        count = 0
+
+        if self._length != other.length():
+            return False
+
+        while count < self._length:
+
+            if self[count] != other[count]:
+                return False
+
+            curr_node = curr_node.next
+            count += 1
+
+        return True
+
+    
+    def __str__(self) -> str:
+        '''
+        pre: 
+            any list
+        post: 
+            returns list as string 
+        '''
+        string = '['
+
+        curr_node = self.head
+
+        while curr_node: # проверка на None
+            string += curr_node.elem + ', ' * (curr_node.next != None) 
+            curr_node = curr_node.next 
+        
+        string += ']'
+
+        return string
+
+    # R
+    def __getitem__(self, index):
+        '''
+        Get element by index
+
+        pre: 
+            index in:
+                [-len; len)
+        ? inv: 
+            index validation 
```

Comment:
It's best to be more wordy in docs than overly succinct, unless there is a well-known convention that developers use to write documentation and everyone understands what this brevity means.

I suppose this `? inv` clause describes what this method does when the precondition doesn't hold.

```suggestion
            raises an `Exception` for index out of bounds
```